service/src/test/java/com.codeforcommunity/JooqMock.java

```diff
@@ -189,14 +195,31 @@ int getCallCount() {
     }
   }
 
-  /**
-   * Constructor for JooqMock. 'UNKNOWN' and 'DROP/CREATE' operations are added by
-   * default.
-   */
+  /** Constructor for JooqMock. 'UNKNOWN' and 'DROP/CREATE' operations are added by default. */
   public JooqMock() {
     // create DSL context
     MockConnection connection = new MockConnection(this);
-    context = DSL.using(connection, SQLDialect.POSTGRES);
+    raw_context = DSL.using(connection, SQLDialect.POSTGRES);
+    context = spy(raw_context);
+    id = 1;
```

Comment:
Magic number?

Reply:
Its initializing the variable. It gets updated (it's not thread safe but we don't really multi-thread a lot so it's fine), so it doesn't stay the same. It's used to set the 'current' ID of an object on insert, and it gets incremented after every insert.